pkg/services/folder/folderimpl/folder.go

```diff
@@ -618,11 +618,11 @@ func (s *Service) getAvailableNonRootFolders(ctx context.Context, q *folder.GetC
 		})
 	} else {
 		dashFolders, err = s.GetFolders(ctx, folder.GetFoldersQuery{
-			UIDs:             folderUids,
-			OrgID:            q.OrgID,
-			SignedInUser:     q.SignedInUser,
-			OrderByTitle:     true,
-			WithFullpathUIDs: true,
+			UIDs:         folderUids,
+			OrgID:        q.OrgID,
+			SignedInUser: q.SignedInUser,
+			OrderByTitle: true,
+			// WithFullpathUIDs: true,
```

Comment:
This code smells a bit, I think this is a good occasion to make it more clear and elegant for everyone. We could write something like this (L609-L627):

```golang
    req := folder.GetFoldersQuery{
			UIDs:         folderUids,
			OrgID:        q.OrgID,
			SignedInUser: q.SignedInUser,
			OrderByTitle: true,
    }
    // Explain why etc.
    if forceLegacy {
      req.WithFullpathUIDs = true
    }
	dashFolders, err = s.GetFolders(ctx, req)
```

Reply:
Truly smelly. We opted for another solution instead see

https://github.com/grafana/grafana/pull/101245